src/openrct2/Context.cpp

```diff
@@ -941,6 +943,36 @@ namespace OpenRCT2
             _uiContext->Update();
         }
 
+        /**
+         * Ensure that the custom user content folders are present
+         */
+        void EnsureUserContentDirectoriesExist()
+        {
+            EnsureDirectoriesExist(
+                DIRBASE::USER,
+                {
+                    DIRID::OBJECT,
+                    DIRID::SAVE,
+                    DIRID::SCENARIO,
+                    DIRID::TRACK,
+                    DIRID::LANDSCAPE,
+                    DIRID::HEIGHTMAP,
+                    DIRID::THEME,
+                    DIRID::SEQUENCE,
+                });
+        }
+
+        void EnsureDirectoriesExist(const DIRBASE dirBase, const std::vector<DIRID> dirIds)
+        {
+            std::string path;
+            for (const auto& dirId : dirIds)
+            {
+                path = _env->GetDirectoryPath(dirBase, dirId);
```

Comment:
You can define path inside the for loop.

Reply:
The intention here was to not have it allocate a new string every iteration of the for-loop, but I forget if that would be an issue in C++ (it's been a little while) :)

Reply:
Yeah, I suppose you are right.

Reply:
I changed it, since i just read that this way the scope is clearer and I guess it might optimize it compiletime :)